### PR TITLE
NoisyDetector subsampling enhancement 

### PR DESCRIPTION
**Pull request by spbos**
This commit adds subsampling capability to the NoisyDetector class. The subsampling factor and statistic parameters are added to __init__() to prevent unexpected behaviour. Also added test code. 

This enhancement was suggested in #75

### CODE REVIEW

File: hcipy/optics/detector.py
Comment (ehpor) [inline hcipy/optics/detector.py:149]: You should make a distinction between `input_grid` (the grid on which the incoming wavefront should be defined) and `detector_grid` (the grid of the camera image). The input grid is now used as the detector grid. Not sure which one to use as the argument for `__init__()`. If input grid is used, you should check whether subsampling can be performed on this grid (in the initializer to catch mistakes early). Using detector grid as the initializer seems the right choice to me, but you should look at how that works in practice for a few minutes to make sure this is actually the better choice.

**Comment by ehpor** [comment]
@spbos What's the progress on this? Any problems you encountered while addressing my comments?

**Comment by spbos** [comment]
I have addressed the comments, but did not have time to push them yet. It will come soon.

**Comment by spbos** [comment]
@ehpor I've addressed your comments a while back.

**Review by ehpor — APPROVED** [review]

**Comment by ehpor** [comment]
I moved the subsampling functionality to the `Detector` base class to also have subsampling in the `NoiselessDetector` class. Still some leftover points, but I'm not gonna wait for those fixes before merging.

* Should the flatfield be applied before or after adding photon noise? Right now it's applied after, but I think it has to be before, acting as variations in the QE.
* QE is still missing (potentially a function of wavelength).
* Read noise should be quantized, right? Right now it isn't.
* I don't like how `weight` is used in the `Detector.integrate()`. Originally, it was intended to be used for polychromatic lightsources, where multiple wavelengths are sequentially integrated on the detector. In order to not add more dark noise than expected, you could add a weight of 1/N to each call of `Detector.integrate()`. This is fairly clunky and potentially wrongly implemented. We might have to sit down for this and discuss it further.
* `NoisyDetector` with default parameters now acts as a `NoiselessDetector` with photon noise, which can be turned off. It might be better to integrate both into a `SimpleDetector` class.

I'll merge once the CI finishes (excluding #79, which will be fixed in a separate PR).

**Pull request merged**
